bigtable/integration_test.go

```diff
@@ -1450,6 +1459,176 @@ func TestIntegration_AdminCreateInstance(t *testing.T) {
+	if err != nil {
+		t.Fatalf("IntegrationEnv: %v", err)
+	}
+	defer testEnv.Close()
+
+	if !testEnv.Config().UseProd {
+		t.Skip("emulator doesn't support instance creation")
+	}
+
+	timeout := 5 * time.Minute
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	iAdminClient, err := testEnv.NewInstanceAdminClient()
+	if err != nil {
+		t.Fatalf("NewInstanceAdminClient: %v", err)
+	}
+	defer iAdminClient.Close()
+
+	adminClient, err := testEnv.NewAdminClient()
+	if err != nil {
+		t.Fatalf("NewAdminClient: %v", err)
+	}
+	defer adminClient.Close()
+
+	table := testEnv.Config().Table
+	clusterID := testEnv.Config().Cluster
+
+	keyRingName := os.Getenv("GCLOUD_TESTS_GOLANG_KEYRING")
+	if keyRingName == "" {
+		t.Fatal("GCLOUD_TESTS_GOLANG_KEYRING must be set. See CONTRIBUTING.md for details")
+	}
+	kmsKeyName := keyRingName + "/cryptoKeys/key1"
+
+	conf := &InstanceWithClustersConfig{
+		InstanceID: instanceToCreate,
+		// InstanceType: PRODUCTION,
+		DisplayName: "test instance",
+		Clusters: []ClusterConfig{
+			{
+				ClusterID:  clusterID,
+				KMSKeyName: kmsKeyName,
+				Zone:       instanceToCreateZone,
+				NumNodes:   1,
+			},
+		},
+	}
+	if err := iAdminClient.CreateInstanceWithClusters(ctx, conf); err != nil {
+		t.Fatalf("CreateInstance: %v", err)
+	}
+	defer iAdminClient.DeleteInstance(ctx, instanceToCreate)
+
+	// Delete the table at the end of the test. Schedule ahead of time
+	// in case the client fails
+	defer deleteTable(ctx, t, adminClient, table)
+	if err := adminClient.CreateTable(ctx, table); err != nil {
+		t.Fatalf("Creating table: %v", err)
+	}
+
+	encryptionKeyVersion := kmsKeyName + "/cryptoKeyVersions/1"
+
+	// The encryption info can take 30-300s (currently about 120-190s) to
+	// become ready.
+	for i := 0; i < 30; i++ {
+		encryptionInfo, err := adminClient.EncryptionInfo(ctx, table)
+		if err != nil {
+			t.Fatalf("EncryptionInfo: %v", err)
+		}
+
+		kmsKeyVersion := encryptionInfo[clusterID][0].KMSKeyVersion
+		if kmsKeyVersion != "" {
+			break
+		}
+
+		time.Sleep(time.Second * 10)
+	}
+
+	// Validate Encryption Info under getTable
+	table2, err := adminClient.getTable(ctx, table)
+	if err != nil {
+		t.Fatalf("Getting Table: %v", err)
+	}
+	if got, want := len(table2.ClusterStates), 1; !cmp.Equal(got, want) {
+		t.Fatalf("Table Cluster States %v, want: %v", got, want)
+	}
+	clusterState := table2.ClusterStates[clusterID]
+	if got, want := len(clusterState.EncryptionInfo), 1; !cmp.Equal(got, want) {
+		t.Fatalf("Table Encryption Info Length: %v, want: %v", got, want)
+	}
+	tableEncInfo := clusterState.EncryptionInfo[0]
+	if got, want := int(tableEncInfo.EncryptionStatus.Code), 0; !cmp.Equal(got, want) {
+		t.Fatalf("EncryptionStatus: %v, want: %v", got, want)
+	}
+	// NOTE: this EncryptionType is btapb.EncryptionInfo_EncryptionType
+	if got, want := tableEncInfo.EncryptionType, btapb.EncryptionInfo_CUSTOMER_MANAGED_ENCRYPTION; !cmp.Equal(got, want) {
```

Comment:
it would be nice to have an assertion for non cmek clusters on the table encryption info